_posts/2025-03-09-no-esp32-style-backdoor.markdown

```diff
@@ -0,0 +1,78 @@
+
+So precisely what *can* a driver access?
+That's what [CHERIoT's auditing flow](https://cheriot.org/book/audit.html) is designed to tell you.
+When you build a CHERIoT firmware image, the linker emits a report explaining everything that every compartment has access to.
+
+In the CHERIoT network stack, the network device is owned by the firewall compartment, which instantiates the network device driver provided by the specific target platform.
+This compartment is trusted for network availability (it can always drop incoming or outgoing network packets) and is trusted for confidentiality and integrity of any unencrypted network traffic.
+
+If you look at the audit report for this compartment you can see that it has access to:
+
+ - A bunch of C library routines (`memcpy`, atomic add, and so on friends).
+ - The MMIO region for the network device.
+ - A capability that authorises it to wait on and acknowledge the interrupt associated with the network device.
+ - Allocator APIs for memory allocation and a capability for a small heap quota.
+ - Scheduler APIs for waiting on a futex.
+ - An API in the DNS compartment to forward DNS replies (and another that initialises the DNS compartment).
+ - An API in the TCP/IP compartment to forward incoming packets that are not DNS responses.
+ 
+This is an exhaustive list.
+Even if the network driver (which is compiled into the firewall compartment) is malicious, this lets you reason about everything that it can do.
+It cannot expose an API to write all memory, because it does not have access to all memory.
+It cannot expose an API to write to flash, because it does not have access to all flash.
+
+These guarantees hold no matter how an attacker gains arbitrary-code execution in that compartment, whether by exploiting an unintentional bug or a back door maliciously introduced in the supply chain.
+
+# How much further does CHERIoT go?
+
+We designed the CHERIoT software model to simultaneously support:
+
+ - Regulatory requirements that a particular device must be accessible to only a specific binary driver.
+ - Proposed right-to-repair legislation requiring users to be able to modify all of the other behaviour on the device.
+ - Strong, auditable, security properties between components provided by different people.
+
+You can build a code signing flow with our auditing tools that signs any firmware as long as the device is accessible only to code with an exact binary hash (which you may generate from SBOM tooling) and, at the same time, add policy checks that limit what that component can access.
+Even if you have to run some binary-only driver, you can ensure that it respects the principle of least privilege and has no access to anything except the device and other things that you explicitly pass to it.
+
+This is what is possible when you design a platform for *usable* security from the ground up, rather than trying to retrofit it to abstractions for scaled-down models of 1970s minicomputers.
```

Comment:
This sentence may need a bit more context, maybe a link to the c is not a high level language paper or something?